sklearn/feature_extraction/text.py

```diff
@@ -133,12 +133,24 @@ def _word_ngrams(self, tokens, stop_words=None):
         min_n, max_n = self.ngram_range
         if max_n != 1:
             original_tokens = tokens
-            tokens = []
+            if min_n == 1:
+                # no need to do any slicing for unigrams
+                # just iterate through the original tokens
+                tokens = list(original_tokens)
+                min_n += 1
+            else:
+                tokens = []
+
             n_original_tokens = len(original_tokens)
+
+            # bind method outside of loop to reduce overhead
+            tokens_append = tokens.append
+            space_join = " ".join
+
             for n in xrange(min_n,
                             min(max_n + 1, n_original_tokens + 1)):
                 for i in xrange(n_original_tokens - n + 1):
-                    tokens.append(" ".join(original_tokens[i: i + n]))
+                    tokens_append(space_join(original_tokens[i: i + n]))
```

Comment:
as a quick glance I would say that most of the time will be spend on the slicing `original_tokens[i: i + n]`, it might be possible to swap the two loops to get something like (roughly)

```
for i in ...:
    _current = ''
    for _tok in original_tokens[i:i + n]:
            _current = _current+' '+_tok
            tokens_append(
                _current
            )
```

Thus only iterating over the slice, which should likely be faster as well.
Would need to be profiled though.